launcher/ui/dialogs/ModDownloadDialog.cpp

```diff
@@ -0,0 +1,98 @@
+#include "ModDownloadDialog.h"
+
+#include <BaseVersion.h>
+#include <icons/IconList.h>
+#include <InstanceList.h>
+
+#include "ProgressDialog.h"
+
+#include <QLayout>
+#include <QPushButton>
+#include <QValidator>
+#include <QDialogButtonBox>
+
+#include "ui/widgets/PageContainer.h"
+#include "ui/pages/modplatform/modrinth/ModrinthPage.h"
+#include "ModDownloadTask.h"
+
+
+ModDownloadDialog::ModDownloadDialog(const std::shared_ptr<ModFolderModel> &mods, QWidget *parent,
+                                     BaseInstance *instance)
+    : QDialog(parent), mods(mods), m_instance(instance)
+{
+    setObjectName(QStringLiteral("ModDownloadDialog"));
+    resize(400, 347);
```

Comment:
This should be in the UI file

Reply:
This dialog was acutally made without a ui file, this is made based off the VersionSelectDialog